pandas/core/window/rolling.py

```diff
@@ -658,9 +658,24 @@ def _numba_apply(
             return self._resolve_output(out, obj)
 
     def aggregate(self, func, *args, **kwargs):
-        result = ResamplerWindowApply(self, func, args=args, kwargs=kwargs).agg()
+        # GH46132
+        _axis_modifed_flag = False
+        # modifying axis and transposing dataframe should not be needed
+        # once ReamplerWindow supports axis = 1
+        if self.axis == 1:
```

Comment:
why are we not using the `_get_data_to_aggregate` pattern here? this seems like making yet another way to do this.

Reply:
Did you mean to write another function or use some methods in some python class to get the data ready for aggregation?

Reply:
`_get_data_to_aggregate` pattern was never implemented on rolling, just groupby. 

Maybe instead @xr-chen, could you try to investigate assigning `ResamplerWindowApply`'s `self.axis` attribute like `GroupByApply` fixes this instead?

```
self.axis = obj.obj._get_axis_number(kwargs.get("axis", 0))
```

Reply:
Thanks for your suggestion. I will have a try.

Reply:
I have tried to initialize `ResamperWindowApply`'s `self.axis` like in `GroupByApply`, but that doesn't work. The `agg_list_like` method of `Apply` class behaves the same whatever the `self.axis` is, and it still calls `obj._gotitem` with `subset` is a `Series` and `self.axis=1` which finally causes an error.